### PR TITLE
Add application sandbox

### PR DESCRIPTION
**Pull request by letmaik**
Implements #1693.

### CODE REVIEW

**Comment by letmaik** [comment]
I should note that this invalidates a bunch of docs but I'd prefer if that's done in a follow-up PR.

File: tests/sandbox/sandbox_gov.lua [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+-- This file defines the default initial contents (ie, Lua scripts) of the governance scripts table.
+return {
+  pass = [[
+  tables, calls, votes = ...
+
+  -- interface definitions
+  PASSED = 1
+  PENDING = 0
+  REJECTED = -1
+
```
Comment (achamayou) [inline tests/sandbox/sandbox_gov.lua:13]: A comment similar to:

The constitution in a real CCF application deployment would at least count votes and compare to a threshold of members, but in the sandbox sample, all votes pass automatically.

**Review by achamayou — APPROVED** [review]

File: tests/sandbox/sandbox.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    START_NETWORK_SCRIPT="${PATH_HERE}"/../start_network.py
+    pip install -q -U -e "${PATH_HERE}"/../../python/
+    pip install -q -U -r "${PATH_HERE}"/../requirements.txt
+fi
+
+echo "Python environment successfully setup"
+
+CURL_CLIENT=ON \
+    python "${START_NETWORK_SCRIPT}" \
+    --enclave-type virtual \
```
Comment (achamayou) [inline tests/sandbox/sandbox.sh:35]: It's unfortunately awkward for a user to override this if they want to run with sgx.

File: tests/sandbox/sandbox.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    --initial-member-count 1 \
+    --initial-user-count 1 \
+    --gov-script "${GOV_SCRIPT}" \
```
Comment (achamayou) [inline tests/sandbox/sandbox.sh:38]: Likewise if they want to test a different governance. I think these two options should allow an override.

File: tests/sandbox/sandbox.sh (hunk 2 above)
Comment (letmaik) [reply tests/sandbox/sandbox.sh:35]: why? just pass --enclave-type debug, later argument override earlier ones

Comment (achamayou) [reply tests/sandbox/sandbox.sh:35]: I did not know you could do that, great!

**Comment by letmaik** [comment]
Running a nightly ci run to be sure I didn't break anything https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=13608&view=results

File: tests/sandbox/sandbox.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+PATH_HERE=$(dirname "$(realpath -s "$0")")
+VERSION_FILE="${PATH_HERE}"/../share/VERSION
+GOV_SCRIPT="${PATH_HERE}"/sandbox_gov.lua
+
+echo "Setting up Python environment..."
+
+if [ ! -f "sandbox_env/bin/activate" ]; then
+    python3.8 -m venv sandbox_env
+fi
+source sandbox_env/bin/activate
```
Comment (eddyashton) [inline tests/sandbox/sandbox.sh:16]: Can we add `ccf_` to this name? Just thinking about the actual user flow of running this from a CCF install, if we're going to add anything to the local directory we should name it clearly. I don't know how common this impromptu venv creation is - should this be under `/tmp`?

**Review by eddyashton — APPROVED** [review]
Can you add something to the `CHANGELOG` in this PR, and open a TODO issue to make sure we remember the docs update?

**Comment by letmaik** [comment]
@eddyashton I created an issue regarding docs update. I think the changelog is probably best updated later and can then also link to the docs.

File: tests/sandbox/sandbox.sh (hunk 4 above)
Comment (letmaik) [reply tests/sandbox/sandbox.sh]: I changed it to `.venv_ccf_sandbox` (the [Python docs](https://docs.python.org/3/tutorial/venv.html#creating-virtual-environments) say that `.venv` is a common name) and allow overriding it by setting the `VENV_DIR` environment variable, if someone really wants to.

File: tests/sandbox/sandbox.sh [hunk 5]
```diff
@@ -15,6 +15,7 @@ echo "Setting up Python environment..."
 if [ ! -f "${VENV_DIR}/bin/activate" ]; then
     python3.8 -m venv "${VENV_DIR}"
 fi
+# shellcheck source=/dev/null
```
Comment (achamayou) [inline tests/sandbox/sandbox.sh:18]: Is this the "source only from hardcoded paths" warning? If so we can turn it off globally I think.

Comment (letmaik) [reply tests/sandbox/sandbox.sh:18]: Yeah, I think turning it off makes sense.

**Pull request merged**